.circleci/config.yml

```diff
@@ -0,0 +1,115 @@
+version: 2.1
+tagged_build_filters: &tagged_build_filters
+  branches:
+    ignore: /.*/
+  tags:
+    only: /v[0-9]+\.[0-9]+\.[0-9]+/
+test_build_filters: &test_build_filters
+  branches:
+    only: /.*/
+  tags:
+    ignore: /v[0-9]+\.[0-9]+\.[0-9]+/
+jobs:
+  test:
+    docker:
+      - image: circleci/ruby:2.7.0
+    steps:
+      - checkout
+      - run: bundle install
+      - run: rubocop -c .rubocop.yml -a
+  build:
+    docker:
+      - image: circleci/ruby:2.7.0
+    steps:
+      - checkout
+      - run:
+          name: Check Tagged Push
+          command: |
+            PKG_VERSION=$(cat lib/version.rb | grep VERSION | cut -d'"' -f2)
+            if [[ "${CIRCLE_TAG}" != "v${PKG_VERSION}" ]]; then
+              echo "There is mismatch:"
+              echo "  TAG_VERSION: ${CIRCLE_TAG}"
+              echo "  PKG_VERSION: v${PKG_VERSION}"
+              exit 1
+            fi
+      - run: gem build logdna-rails.gemspec
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./logdna-rails-*.gem
+  app2pub:
```

Comment:
What does this step actually do? It attaches the workspace and saves the already existing file there? Or is it somehow used for the manual approval step?

Reply:
Yes - previously, I tried on `agent` changing type of the functional job to `approval` but it didn't work - basically, there should be a separate job for approval stage